src/vs/workbench/contrib/scm/common/scmService.ts

```diff
@@ -71,7 +73,23 @@ class SCMInput implements ISCMInput {
 	private readonly _onDidChangeValidateInput = new Emitter<void>();
 	readonly onDidChangeValidateInput: Event<void> = this._onDidChangeValidateInput.event;
 
-	constructor(readonly repository: ISCMRepository) { }
+	constructor(readonly repository: ISCMRepository, @IStorageService private storageService: IStorageService) {
+		this._value = this.storedValue() ?? '';
+	}
+
+	store(value: string) {
+		let root = this.repository.provider.rootUri;
+		if (root) {
+			this.storageService.store(root.path, value, StorageScope.WORKSPACE);
```

Comment:
Using `root.path` isn't a great idea for a couple reasons:
- the storage service keys are shared across the entire workbench, so we should prefix it with something related to SCM
- there can be two different SCM providers with the same root path, we should take `provider.label` also into account

How about:

```suggestion
			const key = `scm/input:${this.repository.provider.label}:${root.path}`;
			this.storageService.store(key, value, StorageScope.WORKSPACE);
```